source/common/http/filter/fault_filter.h

```diff
@@ -82,14 +93,23 @@ class FaultFilter : public StreamDecoderFilter {
   void setDecoderFilterCallbacks(StreamDecoderFilterCallbacks& callbacks) override;
 
 private:
+  void recordAbortsInjectedStats();
+  void recordDelaysInjectedStats();
   void resetTimerState();
   void postDelayInjection();
   void abortWithHTTPStatus();
-  bool matchesTargetCluster();
+  bool matchesTargetUpstreamCluster();
+  bool matchesDownstreamNodes(const HeaderMap& headers);
 
   FaultFilterConfigSharedPtr config_;
   StreamDecoderFilterCallbacks* callbacks_{};
   Event::TimerPtr delay_timer_;
+  std::string downstream_cluster_{};
+
+  std::string delay_percent_key_{"fault.http.delay.fixed_delay_percent"};
```

Comment:
OK, well per above comment, if we are going to generalize and just do one runtime lookup, these don't need to be local variables anyway. They can just all be local variables inside the lookup function.